source/slang/slang-ir-variable-scope-correction.cpp

```diff
@@ -54,65 +54,78 @@ static void debugPrint(uint32_t indent, IRInst* inst, const char* message)
-                    auto breakBlock = loop->getBreakBlock();
-                    debugPrint(indent++, breakBlock, "break block");
-
-                    // If a block is not dominated by the break block, but dominated by the loop header, then the
-                    // instructions in this block are considered to be defined in the loop. We need to search
-                    // the instruction's use sites to see if there is any uses are out of the loop.
-                    if (!dominatorTree->dominates(breakBlock, block))
+                    debugPrint(indent++, getBlock(use->getUser()), "inst's use block is");
+                    if (auto userBlock = getBlock(use->getUser()))
                     {
-                        debugPrint(indent++, block, "block is not dominated by break block");
-                        // traverse all uses of this instruction
-                        for (auto use = inst->firstUse; use; use=use->nextUse)
+                        // If the use site of this instruction is dominated by the break block, it means that the
+                        // instruction is used after the break block, so we need to make that instruction available globally.
+                        // By doing so, we record all the users of this instructions.
+                        if (dominatorTree->dominates(breakBlock, userBlock))
                         {
-                            debugPrint(indent++, getBlock(use->getUser()), "inst's use block is");
-                            if (auto userBlock = getBlock(use->getUser()))
-                            {
-                                // If the use site of this instruction is dominated by the break block, it means that the
-                                // instruction is used after the break block, so we need to make that instruction available globally.
-                                // By doing so, we record all the users of this instructions.
-                                if (dominatorTree->dominates(breakBlock, userBlock))
-                                {
-                                    debugPrint(indent, inst, "inst is defined in loop but used outside of loop");
-                                    debugPrint(indent++, use->getUser(), "The user is");
-                                    instList.add(use->getUser());
-                                }
-                            }
+                            debugPrint(indent, inst, "inst is defined in loop but used outside of loop");
+                            debugPrint(indent++, use->getUser(), "The user is");
+                            instList.add(use->getUser());
                         }
                     }
                 }
-            }
-
-            if (instList.getCount() > 0)
-            {
-                workListMap.add(inst, instList);
+                if (instList.getCount() > 0)
+                {
+                    workListMap.add(inst, instList);
```

Comment:
You added it to the worklist, but we didn't process the worklist?

Reply:
The process is in line 130.